### PR TITLE
End-to-end test suite with accessibility scans

### PR DESCRIPTION
**Pull request by 1Copenut**
### Added
* Selenium test runner, Chromedriver headless
* Basic smoke test for future levelsetting
* e2e test for every component. Each test runs the Axe scanner using `[section 508', 'wcag2a', 'wcag2aa', 'wcag21aa']` rule sets. This ensures we are as compliant as we can be, outside of best practices. Those are guidelines outside the scope of what we're testing.
* Travis.yml update to build Chrome on the CI/CD server and run all tests headlessly

### Reused Code For Plumbing Test Runners
Thank you to @pwolfert for his blueprint branch linked below. I used that code to split the unit and e2e tests along logical fault lines, so we can run one, the other, or both, easily.
* https://github.com/CMSgov/design-system/tree/pwolfert/visual-regression-testing

---
### Screenshot of latest test run. 

There's an odd Node deprecated method, but I can't find where it's being warned from. Any suggestions?

screen shot 2019-01-30 at 10 01 58 am

### CODE REVIEW

**Comment by line47** [comment]
How do I run these tests @1Copenut is there documentation for that?

**Comment by 1Copenut** [comment]
@line47 If you're looking at the branch locally, the following commands should get you going:

1. `$ yarn install`
2. `$ yarn test:e2e`

It'll take a bit to build the /docs folder and all components, then a lot of information will be coming across the console. It's running on Chrome headless, so no browsers will open, just the console output.

**Comment by line47** [comment]
I ran `npm update` on my local branch then ran `yarn test:e2e` and it was getting stopped at the smoke test.  `CMS Design System smoke test › Should click Getting Started and load the /getting-started page`

I'm unsure which getting started it is supposed to be clicking, the left-hand nav or the button `learn how to get started`

screen shot 2019-02-20 at 2 25 31 pm

**Comment by 1Copenut** [comment]
I changed `$ yarn test:e2e:noBuild` to `$yarn test:e2e:skipBuild` to match the feature flag we're passing on to Node. All tests passing now, and the `--watch` command for unit tests restored.

**Comment by line47** [comment]
@bernardwang We'd love to get your 👀 on this particularly the `.travis.yml` file and any other suggestions for improvement you may have.

**Comment by line47** [comment]
@pwolfert do you have time this week to take a look? 🤞

**Comment by pwolfert** [comment]
Oh right, I forgot about this! I'm putting it on my to-do list today to go through it.

**Comment by line47** [comment]
Awesome thank you! @pwolfert 😄

File: packages/core/dist/helpers/constants.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict';
```
Comment (pwolfert) [inline packages/core/dist/helpers/constants.js]: This file and `e2eTestHelpers.js` don't need to be committed

**Comment by line47** [comment]
Thanks for taking a look Patrick!

1. Helpers sound like a good idea to me
2. I also like the idea of having Jest configured such that the `.e2e.test.js` files can live within the individual component directories, and that might be better for organization

**Comment by 1Copenut** [comment]
@pwolfert Thank you for reviewing this PR. I agree with you on the suggestions to refactor and eliminate duplicate code through abstraction.

I'll take you up on your offer to work through some of these refactors. I'm juggling a few other Design System branches, and appreciate the help!

**Comment by pwolfert** [comment]
@1Copenut, maybe you can just look over the changes and make sure you like them? Otherwise I can just merge it in a couple hours or so. I'm not going to add a review since I changed a bunch of it. @line47

**Comment by 1Copenut** [comment]
@pwolfert I like these changes a lot. Feel free to merge anytime, or I'll merge in the morning if still open.

**Pull request merged**